pkg/cafs/reader.go

```diff
@@ -159,6 +170,85 @@ func (r *chunkReader) WriteTo(writer io.Writer) (n int64, err error) {
 
 }
 
+func (r *chunkReader) calculateKeyAndOffset(off int64) (index int64, offset int64) {
+	index = off / int64(r.leafSize)
+	offset = off - (index * int64(r.leafSize))
+	return
+}
+
+func (r *chunkReader) ReadAt(p []byte, off int64) (n int, err error) {
+
+	// Calculate first key and offset.
+	index, offset := r.calculateKeyAndOffset(off)
+	if index >= int64(len(r.keys)) {
+		return 0, nil
+	}
+
+	var read int
+
+	seekHead := func() {
```

Comment:
might this broken out as a separate function and/or could there be a comment regarding its intent?

in particular, i'm curious to hear more about the (presumed) analogy between what this function is doing and [`seek`](http://man.cat-v.org/plan_9/2/seek) calls on file-handles.

Reply:
seek ahead implies that the reader is seeking ahead to the next blob in the sequence.
The goal here is to serve reads from memory as far as possible when streaming a file. This way a read request never blocks on read completing from backend. Will add the comments.